src/main/java/ua/tc/marketplace/service/impl/AdServiceImpl.java

```diff
@@ -89,4 +110,63 @@ public void deleteAd(Long adId) {
   private Ad getAd(Long adId) {
     return adRepository.findById(adId).orElseThrow(() -> new AdNotFoundException(adId));
   }
+
+  private AdAttribute getAdAttribute(Entry<Long, String> entry, Ad finalAd, Category category) {
+    return new AdAttribute(
+        null,
+        finalAd,
+        category.getAttributes().stream()
+            .filter(attribute -> attribute.getId().equals(entry.getKey()))
+            .findFirst()
+            .orElseThrow(() -> new AttributeNotFoundException(entry.getKey())),
+        entry.getValue());
+  }
+
+  private List<AdAttributeRequestDto> parseJsonAdAttributes(CreateAdDto dto) {
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
It's better to create ObjectMapper only once, during service instantiation. Otherwise it will be create on every request